sources/api/netdog/src/net_config/v3.rs

```diff
@@ -70,21 +79,30 @@ impl Validate for NetConfigV3 {
         }
 
         for (name, device) in &self.net_devices {
+            // Bonds / vlans cannot be configured via MAC address
+            if let NetworkDeviceV1::BondDevice(_) | NetworkDeviceV1::VlanDevice(_) = device {
+                ensure!(
+                    !matches!(name, InterfaceId::MacAddress(_)),
+                    error::InvalidNetConfigSnafu {
+                        reason: "bonds and vlans may not be configured using MAC address"
+                    }
+                )
+            };
+
             // Bonds create the interfaces automatically, specifying those interfaces would cause a
             // collision so this emits an error for any that are found
             if let NetworkDeviceV1::BondDevice(config) = device {
                 for interface in &config.interfaces {
-                    // This checks if the insert already found one, which would be a failure
-                    if !interface_names.insert(interface) {
-                        return error::InvalidNetConfigSnafu {
+                    ensure!(
+                        !interface_names.contains(interface),
+                        error::InvalidNetConfigSnafu {
```

Comment:
I don't think this works, now you aren't adding the interface to the HashSet as you find it. So if someone added the same interface to two different bonds, this logic would miss it.

Reply:
Ah good catch - I read right over that!